### PR TITLE
072 std out check

### PR DESCRIPTION
**Pull request by WojciechRynczuk**
Standard output comparison during tests.
Allows for checking the failing cases.

### CODE REVIEW

File: sources/vcdMaker/test/functional/test_001.std [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+Parsed /home/wojtek/devel/vcdMaker/sources/vcdMaker/test/functional/test_001.txt:
```
Comment (endrjux) [inline sources/vcdMaker/test/functional/test_001.std]: You may want to consider removing sensitive information from path.

File: test/functional/merge.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #
 # A set of classes needed to create a vcdMerge specific test.
 #
-# Copyright (c) 2017 vcdMaker team
```
Comment (endrjux) [inline test/functional/merge.py]: Looks like unneeded edit.

File: test/functional/tests.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 #
 # The master class creating test objects.
 #
-# Copyright (c) 2017 vcdMaker team
```
Comment (endrjux) [inline test/functional/tests.py]: Not needed.

File: test/functional/executor.py [hunk 4]
```diff
@@ -103,6 +124,26 @@ def is_gold_and_output_equal(self):
...
+    def is_standard_output_equal(self):
+        """Compares the standard output with the reference file.
+
+        Returns:
+        True if the standard output and the reference file are equal, false otherwise.
+        """
+
+        line_number = 1
+        with open(self.stdout_filename) as stdout_ref, open(self.stdout_output) as stdout_out:
+            for stdout_line, stdref_line in itertools.zip_longest(stdout_out, stdout_ref):
```
Comment (endrjux) [inline test/functional/executor.py:136]: How does this handle difference in paths? Is this regex responsible for that?

File: sources/vcdMaker/test/functional/test_001.std (hunk 1 above)
Comment (WojciechRynczuk) [reply sources/vcdMaker/test/functional/test_001.std]: Correct, the path will be shortened to just the file name.

File: test/functional/merge.py (hunk 2 above)
Comment (WojciechRynczuk) [reply test/functional/merge.py]: I will revert the previous version.

File: test/functional/tests.py (hunk 3 above)
Comment (WojciechRynczuk) [reply test/functional/tests.py]: I will revert the previous version.

File: test/functional/executor.py (hunk 4 above)
Comment (WojciechRynczuk) [reply test/functional/executor.py:136]: Yes, the regex removes the path leaving just the file name. E.g. `/some/path/to/the/filename` will be reduced to `filename`. Still it is better not to keep the whole path in the reference files as you noticed above. Anyway, this shortening of paths will be left here.

Comment (endrjux) [reply test/functional/executor.py:136]: Why not `os.path.basename()` then?

Comment (WojciechRynczuk) [reply test/functional/executor.py:136]: The truth is that the algorithm doesn't know when it is dealing with the path name. It is just searching for some text that looks like a path (just `/some/text/between/slashes/is/treated/as/a/path/`). When found, the path is removed. The whole file is treated as a blob of text. To use `os.path.basename()` we would have to find something that looks like a path and a filename and then use the function.

Comment (WojciechRynczuk) [reply test/functional/executor.py:136]: Just added also handling Windows paths. As mentioned before we cannot use `os.path.basename()` as the standard output of the application differs heavily fro case to case and we never know when we deal with a path. So the output text is stripped from anything that resembles a path name.

Comment (endrjux) [reply test/functional/executor.py:136]: But we know how vcdmaker outputs stuff. If stdout line begins with "Parsed " then we know the next thing is a path and we could use that knowledge. If you check if both stdout lines begin with "Parsed " then you can just continue. Who said you have to blindly compare stdouts line by line, we know our stdout, right?

Comment (WojciechRynczuk) [reply test/functional/executor.py:136]: Unfortunatelly there are more different contexts (standard output and errors). In the future there might be even more.

Current examples.

Basic one:
```
Parsed /some/path/here/test_003.txt
```

Merging error:
```
[Error 0002]: Inconsistent signal: A.Val. Types: wire / wire. Sizes: 8 / 9. Sources: /some/path/here/test_007.txt and /other/path/here/test_007.txt. At line 2.
```

Merging source description:
```
U{../some/path/here/vcdMaker.xml},71050601,us,System1,Counter,system1_time1.txt
```

I even thoght about putting paths and file names inside some markers but I do not like the idea either. What do you think about it?

Comment (WojciechRynczuk) [reply test/functional/executor.py:136]: I just tried to prepared regexes for all the cases to filter out path names. The regexes get so complicated that I'd rather stick with the current solution.

Comment (endrjux) [reply test/functional/executor.py:136]:
```
for expected_line in expected_output and actual_line in actual_output do
    if actual_line begins with "Parsed " and expected_line begins with "Parsed " then
        continue loop
    else
        fail test

    if actual_line != expected_line then
        fail test
```
Isn't this sufficient?

Comment (WojciechRynczuk) [reply test/functional/executor.py:136]: As we discussed there were more contexts in which the path and the file name appeared. The current solution seems just reasonable.

**Pull request merged**